source/particle/generator/probability_density_function.cc

```diff
@@ -55,6 +55,26 @@ namespace aspect
           {
             Functions::ParsedFunction<dim>::declare_parameters (prm, 1);
 
+            // This parameter overwrites one of the parameters in ParsedFunction
+            // with a new default value of 1.0. The original default value was 0
+            // everywhere, which is not allowed.
+            prm.declare_entry("Function expression",
+                              "1.0",
+                              Patterns::Anything(),
+                              "The formula that denotes the function you want to evaluate for "
+                              "particular values of the independent variables. This expression "
+                              "may contain any of the usual operations such as addition or "
+                              "multiplication, as well as all of the common functions such as "
+                              "`sin' or `cos'. In addition, it may contain expressions like "
+                              "`if(x>0, 1, -1)' where the expression evaluates to the second "
+                              "argument if the first argument is true, and to the third argument "
+                              "otherwise. For a full overview of possible expressions accepted "
+                              "see the documentation of the muparser library at http://muparser.beltoforion.de/."
+                              "\n\n"
+                              "If the function you are describing represents a vector-valued "
+                              "function with multiple components, then separate the expressions "
+                              "for individual components by a semicolon.");
```

Comment:
I think this text (which I know you only copied from the original parameter) is not a good description here because we're considering probability distributions. They can not be negative (with respect to the `if` expression), and they are always scalar (with respect to that last paragraph).

What do you think of the following:
```suggestion
                              "The formula that denotes the spatially variable probability density function. "
                              "This expression "
                              "may contain any of the usual operations such as addition or "
                              "multiplication, as well as all of the common functions such as "
                              "`sin' or `cos'. In addition, it may contain expressions like "
                              "`if(x>0, 1, 0)' where the expression evaluates to the second "
                              "argument if the first argument is true, and to the third argument "
                              "otherwise; this example would result in no particles at all in that "
                              "part of the domain where $x\le 0$, and a constant particle density "
                              "in the rest of the domain. For a full overview of possible expressions accepted "
                              "see the documentation of the muparser library at http://muparser.beltoforion.de/.");
```